modules/tracking/src/gtrTracker.cpp

```diff
@@ -47,7 +47,11 @@ namespace cv
 
 TrackerGOTURN::Params::Params(){}
 
-void TrackerGOTURN::Params::read(const cv::FileNode& /*fn*/){}
+void TrackerGOTURN::Params::read(const cv::FileNode& fn)
+{
+    modelBin = (String)fn["caffemodel"];
+    modelTxt = (String)fn["prototxt"];
```

Comment:
First check that fields are exist

Reply:
Like this:
```cpp

void TrackerGOTURN::Params::read(const cv::FileNode& fn)
{
  if (fn)
  {
    modelBin = (String)fn["caffemodel"];
    modelTxt = (String)fn["prototxt"];
  }
}
```
?

Reply:
the fields are required so we should throw an assertion. Try this:
```cpp
CV_Assert(!fn["caffemodel"].empty()); 
CV_Assert(!fn["prototxt"].empty()); 
```

Reply:
OK